### DISCUSSION TITLE
Which Verify.MSTest usings

### DISCUSSION THREAD

**Discussion by TomW-Skyline** [VerifyTests/Verify#587] [Q&A] [answered]
We are trying to use the Verify.MSTest  NuGet package in a .NET 4.7.2 unit test project. After adding the latest version of the package to the project, we also added `<ImplicitUsings>enable</ImplicitUsings>` to the csproj file as described in the readme.

But unfortunally we keep getting the error that the "Verify()" method cannot be found. "The name Verify does not exist in the current context". In which class is this method defined? Are we missing some usings on top of the file? Which usings are required to use Verify in a .NET 4.7.2 project?

(question also asked on https://stackoverflow.com/questions/73235687/verify-mstest-usings)

**Comment by SimonCropp** [comment, marked as answer]
Add `<ImplicitUsings>enable</ImplicitUsings>` to the project

https://github.com/VerifyTests/Verify#usage

**Reply by TomW-Skyline** [reply]
We did but that doesn't work. ImplicitUsings is only supported since .NET6/C#10.

**Reply by SimonCropp** [reply]
ImplicitUsings are supported in any .net version. but yes u need c#10

If ImplicitUsings are not enabled, substitute usages of `Verify()` with `Verifier.Verify()`.

**Reply by SimonCropp** [reply]
@TomW-Skyline did you downvote my answer on stackoverflow?

**Reply by TomW-Skyline** [reply]
There is no answer yet on Stackoverflow.

**Reply by TomW-Skyline** [reply]
@SimonCropp thank you for the quick answer, but unfortunately we still cannot get it working. The same question remains: which using do we need to add in order to be able to use the "Verifier" class?
`using VerifyMSTest` doesn't work.

![image](https://user-images.githubusercontent.com/97225679/184814079-72b273fa-0e3a-4a51-96ca-c4b7c8df1b53.png)

I've quickly created a repo that reproduces the problem here: https://github.com/TomW-Skyline/VerifyTest-Experiment

**Reply by SimonCropp** [reply]
yep sorry. i forgot mstest requires a base class. see here https://github.com/VerifyTests/Verify#mstest

i did a PR for u https://github.com/TomW-Skyline/VerifyTest-Experiment/pull/1

**Reply by TomW-Skyline** [reply]
@SimonCropp thank you! Had indeed overlooked the base class in the example.
What if the test class already has a base class?

**Reply by SimonCropp** [reply]
> What if the test class already has a base class?

then make that class be based on `VerifyBase`

Happy to accept a PR that adds a static API support for MSTest.

**Reply by TomW-Skyline** [reply]
https://github.com/VerifyTests/Verify/pull/601
